mocksipipeline/instrument/optics/response.py

```diff
@@ -319,3 +323,71 @@ def get_wcs(self, observer, roll_angle=90 * u.deg):
             pc_matrix=pc_matrix,
             observer=observer,
         )
+
+    def psf(self, wavelength=None, mask_oversample=4, fft_oversample = 4):
+        mask = self.aperture.mask(oversample=mask_oversample)
+        f = self.design.focal_length
+
+        x = mask.coords['x'].values * u.micron
+        y = mask.coords['y'].values * u.micron
+        x_step = x[0, 1] - x[0, 0]
```

Comment:
Doesn't this assume that the array is evenly sampled in the same way in each direction?